libvncclient/rfbproto.c

```diff
@@ -1586,6 +1589,31 @@ SendKeyEvent(rfbClient* client, uint32_t key, rfbBool down)
 }
 
 
+/*
+ * SendExtendedKeyEvent.
+ */
+
+rfbBool
+SendExtendedKeyEvent(rfbClient* client, uint32_t keysym, uint32_t keycode, rfbBool down)
+{
+  rfbQemuExtendedKeyEventMsg ke;
+
+  // FIXME: rfbQemuEvent also covers audio events, but this model for checking
+  // for supported messages is somewhat limited, so I'll leave this as is for
+  // now.
+  if (!SupportsClient2Server(client, rfbQemuEvent))
```

Comment:
I'd say that simply returning FALSE here should be the way to go as it not leads any users into thinking their extended key event was sent with the keycode while it it was actually not.

Reply:
Sure. I was just thinking that users will always fall back to regular key events anyhow, and the keysym is already included in the arguments.

Btw, why does `SendKeyEvent` return `TRUE` when the event is not supported?

Reply:
No idea :-) That's apparently plain wrong.